src/frontend/js/components/StepBreadcrumb/index.tsx

```diff
@@ -0,0 +1,101 @@
+// StepBreadcrumb has been created to work in pair with useStepManager hook.
+// Within a step process, it aims to guide the user to know where he/she is in
+// the current process by translating visually the steps manifest.
+import { Fragment } from 'react';
+import { Nullable } from 'types/utils';
+import { Manifest, Step } from 'hooks/useStepManager';
+
+interface Props {
+  manifest: Manifest;
+  step: Nullable<string>;
+}
+
+type FlattenStep = [string, Step];
+
+/**
+ * Retrieve the index of the active step.
+ * When step is null, we know that process is terminated so we have to return
+ * the last step index.
+ * @params step[]
+ * @returns {number} active step index
+ */
+function getActiveStepIndex(steps: FlattenStep[], step: Props['step']) {
+  if (step === null) {
+    return steps.length;
+  }
+  return steps.findIndex((s) => s[0] === step);
+}
+
+/**
+ * The first step of the manifest could be defined as the only step which is
+ * never present in `next` property of another step.
+ * So this method searches which step is not used by another step.
+ * @param {Manifest} manifest
+ * @returns {Manifest.step} step
+ */
+function findFirstStep(manifest: Manifest) {
```

Comment:
```suggestion
function findFirstStep(manifest: Manifest) {
```
I find it confusing not to use directly the starting step declared In the manifest instead of searching for the step that doesn't contain the next step. As we have our first step declared, why not just use it ?

Reply:
Because the `start` property could be any step not necessary the first step of the manifest. In this way, we are able to instantiate the step manager at a given step.

Reply:
ok start step doesn't mean first, I see. It wasn't clear to me but it makes sense. 
Simple question, shouldn't we delegate these properties such as .first like we have current to the manager instead of the component so that the component has a ready-to-use and ordered breadcrumb?

Reply:
Hmmm... I don't think so as step order is only important for the display not for the manager.

Reply:
sorry to insist :-), in the manager, we have the current, the next, the reset, are you sure to get the first() wouldn't have his place here? I find it weird to have to calculate in a component outside the manager using the method next()
I would even use the manager to get a specific step, anything to deal with and ease the use of the manifest could be plugged in the manager.